aziotctl/aziotctl-common/src/config/apply.rs

```diff
@@ -207,7 +208,10 @@ pub fn run(
             super_config::ProvisioningType::None => aziot_identityd_config::ProvisioningType::None,
         };
 
-        aziot_identityd_config::Provisioning { provisioning }
+        aziot_identityd_config::Provisioning {
+            provisioning,
+            local_gateway_hostname,
```

Comment:
See my other comment about X509 auth types and nested mode. It is likely we need to validate the super-config for avoiding nested mode configuration with a device identity that uses X509 auth type (DPS or IoT Hub). I believe IoT Hub X509 auth type will be supported soon however (check with @huguesBouvier)